lib/AST/Printer/TypePrinter.cpp

```diff
@@ -924,17 +928,27 @@ void TypePrinter::printAdjusted(const clang::AdjustedType *T,
 
   // Print the original type, as that is more reflective of what is actually
   // in the code.
-  printBeforeAfter(T->getOriginalType(), OS, std::move(IdentFn));
+  if (tokens.ppa->ShouldPrintOriginalTypeOfAdjustedType()) {
+    printBeforeAfter(T->getOriginalType(), OS, std::move(IdentFn));
+
+  } else {
+    printBeforeAfter(T->getAdjustedType(), OS, std::move(IdentFn));
+  }
 }
 
 void TypePrinter::printDecayed(const clang::DecayedType *T,
                                raw_string_ostream &OS,
                                std::function<void(void)> IdentFn) {
   TokenPrinterContext ctx(OS, T, tokens);
 
-  // Print the decayed representation, otherwise the adjustment will be
-  // invisible.
-  printBeforeAfter(T->getDecayedType(), OS, std::move(IdentFn));
+  // Print the original type, as that is more reflective of what is actually
+  // in the code.
+  if (tokens.ppa->ShouldPrintOriginalTypeOfDecayedType()) {
```

Comment:
This is not a perfect copy. Here we ask the printing policy adaptor if we should print originals of decayed types, rather than of the (more general) adjusted types.

Reply:
Well it differs on just the content of the `if` condition right?
So really it should be `shared_thing( cond_body, T, OS, std::move( IdentFn ) )`.

Reply:
But it's only done twice lol, and both bodies are relatively short. No thanks.